server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsNodeResponse.java

```diff
@@ -116,6 +145,16 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         nodeInfo.writeTo(out);
         nodeStats.writeTo(out);
-        out.writeArray(shardsStats);
+        if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
+            if (aggregatedNodeLevelStats != null) {
+                out.writeOptionalArray(null);
+                out.writeOptionalWriteable(aggregatedNodeLevelStats);
+            } else {
+                out.writeOptionalArray(shardsStats);
+                out.writeOptionalWriteable(null);
```

Comment:
why are you writing this?

Reply:
The decision to return aggregatedNodeLevelStats determines whether to return the aggregated information or the whole list of ShardStats as a whole. We only create the aggregated stats during initialisation [here](https://github.com/opensearch-project/OpenSearch/blob/10a78f7476f41a53dba3a05bb473b61192872ea0/server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsNodeResponse.java#L102-L104)

Writing shardsStats only when the request comes in with aggregatedNodeLevelStats set to `false` implying `aggregatedNodeLevelStats` is false.